pkg/token/server.go

```diff
@@ -289,6 +289,37 @@ func newHandlerFunc(d *daemon, timeout time.Duration) http.Handler {
 		io.WriteString(w, string(response))
 	}
 
-	// timeout handler
-	return http.TimeoutHandler(http.HandlerFunc(mainHandler), timeout, "Handler timeout by token-server-timeout")
+	// logging && timeout handler
+	return withLogging(http.TimeoutHandler(http.HandlerFunc(mainHandler), timeout, "Handler timeout"))
+}
+
+// withLogging wraps handler with logging
+func withLogging(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		startTime := time.Now()
+		log.Infof("Received request: Method[%s], URI[%s], RemoteAddr[%s]", r.Method, r.RequestURI, r.RemoteAddr)
+
+		wrappedWriter := newLoggingResponseWriter(w)
```

Comment:
```suggestion
		// wrap ResponseWriter to cache status code
		wrappedWriter := newLoggingResponseWriter(w)
```
add reason

Reply:
@WindzCUHK fixed in https://github.com/AthenZ/k8s-athenz-sia/pull/83/commits/c89b74a1dfe76a7b34fc0a771676a994955246cb

Reply:
@WindzCUHK added in https://github.com/AthenZ/k8s-athenz-sia/pull/83/commits/2b944d9cd54a3395b49aee7e53a8e7092a12f0e3